crl/x509/crl.go

```diff
@@ -0,0 +1,407 @@
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package x509 parses X.509-encoded keys and certificates.
+package x509
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"io"
+	"math/big"
+	"time"
+
+	"golang.org/x/crypto/cryptobyte"
+	cryptobyte_asn1 "golang.org/x/crypto/cryptobyte/asn1"
+)
+
+var (
+	oidExtensionReasonCode = []int{2, 5, 29, 21}
+)
+
+// RevokedCertificate contains the fields used to create an X.509 v2 Revoked
+// Certificate, i.e. an entry in a Certificate Revocation List's sequence of
```

Comment:
```suggestion
// RevokedCertificate contains the fields used to create an entry in the revokedCertificates
// list of a CRL.
```

X.509 v2 by itself would refer to a certificate. "X.509 v2 CRL" is the terminology used by PKIX to talk about CRLs. CRLs were defined in X.509 v2; they are also defined in X.509 v3, but the version number in the CRL data structure did not change, presumably because the data structure did not change. On the whole, let's omit "X.509 v2" in our docs (including below). It's a bit redundant since we're in the x509 package, and from context it's clear we're talking about CRLs.

Also: Revoked Certificate is not a defined term so we shouldn't use it capitalized.